Neo4j.Driver/Neo4j.Driver/Internal/Extensions/CollectionExtensions.cs

```diff
@@ -15,12 +15,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 using System.Collections;
+using System.Collections.Generic;
 using System.Linq;
+using Neo4j.Driver.V1;
 
 namespace Neo4j.Driver.Internal
 {
     internal static class CollectionExtensions
     {
+        public static T[] DequeueToArray<T>(this Queue<T> queue, int length)
```

Comment:
This one will remove the item from the queue too. Queue.ToArray will just get a copy of that it has without removing any